include/net/net_pkt.h

```diff
@@ -437,15 +437,41 @@ static inline void net_pkt_set_vlan_tag(struct net_pkt *pkt, u16_t tag)
 	pkt->vlan_tci = net_eth_vlan_set_vid(pkt->vlan_tci, tag);
 }
 
+static inline u8_t net_pkt_vlan_priority2pcp(u8_t priority)
```

Comment:
After looking the VLAN code more, there is actually a function called net_vlan2priority() (found in include/net/net_ip.h) that is suppose to do this mapping, but currently it is just doing 1:1 mapping. Everything might actually work as expected without any other modifications if we move this priority2pcp map table there.

Reply:
Thanks! It seems you are right, and PCP is actually just called VLAN priority in the network stack, it seemed ambiguous at first, but the conversion was definitely made in a wrong callback.

I moved the code to the function you suggested. I've also added a sanity-check to the argument (thus a new include). I've already pushed the changes - it should work but I won't be able to verify that on HW until Monday.